R/derive_locf_records.R

```diff
@@ -179,7 +205,7 @@ derive_locf_records <- function(dataset,
   # dataset (with non-missing analysis_var)
   advs_exp_obsv3 <- exp_obsv %>%
     mutate(!!tmp_dtype := "LOCF") %>%
-    anti_join(advs_unique_original, by = c(exp_obs_vars))
+    anti_join(advs_unique_original, by = c(exp_obs_by_vars))
 
   # Merge the expected observations with the input dataset (with non-missing analysis_var)
```

Comment:
I think it is clearer to update the comment:
```suggestion
  # Add the expected observations to the input dataset (with non-missing analysis_var)
```
and change
```
  aval_not_missing_locf <- aval_not_missing %>%
    full_join(advs_exp_obsv3, by = c(exp_obs_vars))
```
to
```
  aval_not_missing_locf <- bind_rows(aval_not_missing, advs_exp_obsv3)
```
Then `exp_obs_vars` could be removed from the code.

Reply:
I've slightly changed the logic of the function, where `exp_obs_vars` is required. And `bind_rows()` was not giving the expected output. But I've updated 'Merge' to 'Add'.